gradle/scripts/lib/scala.gradle

```diff
@@ -58,12 +58,10 @@ configure(scala212 + scala213 + scala3) {
 
 configure(scala212) {
     dependencies {
-        implementation('org.scala-lang:scala-library') {
-            version {
-                strictly '2.12.15'
-            }
+        implementation('org.scala-lang:scala-library:2.12.15')
+        if (managedVersions.containsKey('org.scalameta:munit_2.12')) {
```

Comment:
> We are going to eventually remove `managedVersions`, right?

I initially thought so but now I don't think we can remove `managedVersions`. Because we have to enforce users to use the alias defined at `gradle-scripts`.
It is convenient for users to use any name without strong naming conventions.

Reply:
That makes sense. 😄 
After I read [this](https://github.com/line/armeria/pull/4353/files#diff-d984c6b3a5597bcee9d1c32ba3261f2bdfbef1d2d7b0c4aa68686ff5fdda9c21R196), I thought we are going to remove that eventually, but it's not. 😄

Reply:
`managedVersions` has a known limitation but we need to use it until we find a better answer. 😅